engine.py

```diff
@@ -1084,18 +1112,23 @@ def _get_dialog_parent(self):
         show_dialog & show_modal.
         """
         # determine the parent widget to use:
-        from tank.platform.qt import QtGui, QtCore
-
-        if not self._DIALOG_PARENT:
-            if sys.platform == "win32":
-                # for windows, we create a proxy window parented to the
-                # main application window that we can then set as the owner
-                # for all Toolkit dialogs
-                self._DIALOG_PARENT = self._win32_get_proxy_window()
-            else:
-                self._DIALOG_PARENT = QtGui.QApplication.activeWindow()
-            
-        return self._DIALOG_PARENT
+        try:
```

Comment:
do we need this big catch-all exception trap around window parenting? Can we just do it round the windows logic? Can we catch a specific exception type rather than `Exception`?

Reply:
Good catch. That was a temporary hack while I was debugging that I forgot to remove!